rustsat/src/types.rs

```diff
@@ -697,6 +701,41 @@ impl Assignment {
             Some(var![self.assignment.len() as u32 - 1])
         }
     }
+
+    //Read a solution from a SAT solver's output
+    pub fn from_solver_output_path<P: AsRef<Path>>(path: P) -> anyhow::Result<Self> {
+        let reader = instances::fio::open_compressed_uncompressed_read(path)
+            .context("failed to open reader")?;
+        match instances::fio::parse_sat_solver_output(std::io::BufReader::new(reader)) {
+            Ok(instances::fio::SolverOutput::Sat(assignment)) => Ok(assignment),
+            _ => Err(anyhow::anyhow!(
+                instances::fio::SatSolverOutputError::Nonsolution
+            )), // return some error
+        }
+    }
+
+    /// Parses and saves literals from value line.
+    pub fn from_vline(&mut self, line: &str) -> anyhow::Result<()> {
+        for number in line.split(' ') {
+            let mut number_v = 0;
+            if number.parse::<i32>().is_ok() {
+                number_v = number.parse::<i32>().unwrap();
+            }
+            //End of the value lines
+            if number_v == 0 {
+                continue;
+            }
+
+            let literal = match Lit::from_ipasir(number_v) {
+                Ok(lit) => lit,
+                Err(error) => return Err(anyhow::anyhow!(error)),
+            };
+
+            self.assign_lit(literal);
+        }
+
+        return Ok(());
```

Comment:
I'd prefer this to be a constructor following Rust's `from_` naming scheme
```suggestion
    pub fn from_vline(line: &str) -> anyhow::Result<Self> {
        // skip `v ` tag
        let line = line[2:];
        let assignment = Assignment::default();
        for number in line.split(' ') {
            let number = number.parse::<i32>()?;
            //End of the value lines
            if number == 0 {
                return assignment;
            }
            let literal = Lit::from_ipasir(number)?;
            assignment.assign_lit(literal);
        }

        return Ok(());
```